### PR TITLE
refactor: Refactored feedback POST request

### LINKED ISSUE #14: Refactor Feedback API
**Describe the issue**
We are improving the reliability and reusability of our code. Thereby, we need to simplify and document our code. Use good DRF practices taking <a href="https://github.com/ECellNitrr/teesco-backend">teesco-backend</a> as reference.

Navigate to `feedback/views.py` and refer `feedback` view and simplify it. Also write its swagger documentation.

**Expected behaviour**
Code must be understandable and reproducible.

### PR DESCRIPTION
**Pull request by sahil9001**
### Description
Refactored and changed the api_view wo APIView with swagger implementation and responses.

**Type of Change:** (Delete irrelevant options)
- Code
- Documentation

**Code/Quality Assurance Only** (Delete irrelevant options)

- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Locally tested

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [x] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: feedback/responses.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+feedback_created_201 = {
+    "message" : "Feedback posted successfully"
+}
+
+feedback_invalid_400 = {
+    "message" : "error_msg"
```
Comment (anshsrtv) [inline feedback/responses.py:6]: Specify the error messages that can be returned as response from the API. The `error_msg` is a variable.

File: feedback/urls.py [hunk 2]
```diff
@@ -1,7 +1,6 @@
 from django.contrib import admin
```
Comment (anshsrtv) [inline feedback/urls.py:1]:
```suggestion
```

File: feedback/views.py [hunk 3]
```diff
@@ -1,29 +1,30 @@
+from rest_framework import response
+from utils.swagger import set_example
 from django.shortcuts import render
+from drf_yasg.utils import swagger_auto_schema
 from rest_framework.decorators import api_view
```
Comment (anshsrtv) [inline feedback/views.py:5]:
```suggestion
from utils.swagger import set_example
from drf_yasg.utils import swagger_auto_schema
```

File: feedback/views.py [hunk 4]
```diff
@@ -1,29 +1,30 @@
...
-        res_status = status.HTTP_400_BAD_REQUEST
-    else:
-        f.save()
-        res_message = "Feedback Posted Successfully"
-        res_status = status.HTTP_200_OK
-
-    return Response({
-        "message": res_message
-    }, status=res_status)
+from rest_framework import serializers, status
```
Comment (anshsrtv) [inline feedback/views.py:9]:
```suggestion
from rest_framework import status
```

**Review by anshsrtv — CHANGES_REQUESTED** [review]
Nice Job 🎉 

- Ensure readability of code. 
- I posted `string` as email and it was posted !!
- Remove unwanted code.

File: feedback/responses.py (hunk 1 above)
Comment (sahil9001) [reply feedback/responses.py:6]: Actually it could be any validation error, there are many options, should I specify any one of them here?

Comment (anshsrtv) [reply feedback/responses.py:6]: Try specifying most of them in a list 👍

File: feedback/views.py (hunk 4 above)
Comment (sahil9001) [reply feedback/views.py]: Done sir

File: feedback/views.py (hunk 3 above)
Comment (sahil9001) [reply feedback/views.py]: Done sir

File: feedback/urls.py (hunk 2 above)
Comment (sahil9001) [reply feedback/urls.py:1]: Done sir

File: feedback/responses.py (hunk 1 above)
Comment (sahil9001) [reply feedback/responses.py]: Done sir

File: feedback/views.py [hunk 5]
```diff
@@ -1,29 +1,27 @@
...
+class FeedbackView(APIView):
+    @swagger_auto_schema(
+        request_body=FeedbackSerializer,
+        responses={
+            '201' : set_example(responses.feedback_created_201),
+            '400' : set_example(responses.feedback_invalid_400)
+        }
+    )
+    def post(self,request):
+        f = FeedbackSerializer(data=request.data)
```
Comment (priyanshisharma) [inline feedback/views.py:17]: This does the job too, though I'll recommend using a better variable name than `f`.

File: feedback/views.py [hunk 6]
```diff
@@ -1,29 +1,27 @@
...
+        if f.is_valid():
+            f.save()
+            return Response({"message" : "Feedback Posted Successfully"},status.HTTP_201_CREATED)
```
Comment (priyanshisharma) [inline feedback/views.py:20]:
```suggestion
            return Response(responses.feedback_created_201, status.HTTP_201_CREATED)
```
You might like to use the responses itself.

File: feedback/views.py [hunk 7]
```diff
@@ -1,29 +1,27 @@
...
+        else:
```
Comment (priyanshisharma) [inline feedback/views.py:21]: This `else` is redundant and can be eliminated.

**Review by priyanshisharma — CHANGES_REQUESTED** [review]
It looks and works well, kudos! 🙌 
I've added a few suggestions, and would also recommend linting your code. Have a look.

File: feedback/views.py (hunk 6 above)
Comment (sahil9001) [reply feedback/views.py:20]: Ohh I see XD

Comment (priyanshisharma) [reply feedback/views.py]: Yesss 😂✌️

**Review by priyanshisharma — APPROVED** [review]
Looks great to me! 🎉

**Review by anshsrtv — CHANGES_REQUESTED** [review]
Change line 6 in `feedback/models.py` to `email = models.EmailField()` 
Make migrations for the same and push them too. ✌️

**Comment by sahil9001** [comment]
@anshsrtv ?

**Review by anshsrtv — APPROVED** [review]

**Pull request merged**